src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php

```diff
@@ -86,6 +88,16 @@ protected function instantiateController($class)
             return $this->container->get($class);
         }
 
-        return parent::instantiateController($class);
+        $e = null;
+
+        try {
+            return parent::instantiateController($class);
+        } catch (FatalThrowableError $e) {
+        } catch (\ArgumentCountError $e) {
+        }
+
+        if ($this->container instanceof Container && in_array($class, $this->container->getRemovedIds())) {
+            throw new \LogicException(sprintf('Controller "%s" cannot be fetched from the container because it is private. Did you forget to tag the service with "controller.service_arguments"?', $class), 0, $e);
```

Comment:
It looks to weird from a DX POV to recommend to add this tag instead of just recommending to make this service public.

Reply:
I crafted the message so that, from DX pov, the best experience is suggested by default (allowing services as arguments, ie the tag)
and from a more exprienced DX pov, the word "private" hints the advanced dev about "public" being just missing.
If doesn't know the difference, then we should definitely suggest the tag. Enhance the current message.